src/appConfigurationImpl.ts

```diff
@@ -627,23 +636,23 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
 
         // try refresh if any of watched settings is changed.
         let needRefresh = false;
-        let changedSentinel;
-        let changedSentinelWatcher;
+        let changedSentinel: WatchedSetting | undefined;
+        let changedSentinelWatcher: SettingWatcher | undefined;
         if (this.#watchAll) {
             needRefresh = await this.#checkConfigurationSettingsChange(this.#kvSelectors);
         } else {
             for (const watchedSetting of this.#sentinels.keys()) {
                 const configurationSettingId: ConfigurationSettingId = { key: watchedSetting.key, label: watchedSetting.label, etag: this.#sentinels.get(watchedSetting)?.etag };
-                const response = await this.#getConfigurationSetting(configurationSettingId, {
-                    onlyIfChanged: true
-                });
-
-                const watcher = this.#sentinels.get(watchedSetting);
-                if (response?.statusCode === 200 // created or changed
-                    || (response === undefined && watcher?.etag !== undefined) // deleted
-                ) {
+                const response: GetConfigurationSettingResponse | undefined =
```

Comment:
This method `refreshKeyValues` contains lots of changes but nothing is related to AFD, right?
I see 3 logic changes here:
1. The way we detect sentinel etag change (`isChanged`) by comparing etag
2. The way we set `changedSentinelWatcher` when sentinel is deleted 
3. Retry opportunity on failed refresh (`if (changedSentinel && changedSentinelWatcher)`)


All these look like bug fixes that should've been done in a separate PR to keep this PR AFD related only.